app/models/katello/events/publish_component_composites.rb

```diff
@@ -0,0 +1,29 @@
+module Katello
+  module Events
+    class PublishComponentComposites
+      EVENT_TYPE = 'publish_component_composites'.freeze
+
+      attr_reader :version
+
+      def initialize(cvv_id)
+        @version = ::Katello::ContentViewVersion.find_by_id(cvv_id)
+      end
+
+      def run
+        return unless version
+
+        description = _("Auto Publish - Triggered by '%s'") % version.name
+        version.content_view.auto_publish_components.each do |component|
+          begin
+            ForemanTasks.sync_task(::Actions::Katello::ContentView::Publish,
```

Comment:
should this be a sync_task?  that'd cause the event queue to be held up until the publish is complete

Reply:
oh i see, this is the reason its more reliable :)
I kinda envisioned it a tad differently:

1)  the composite id is put on the queue so that way its a single task that need to get started
2) The tasks are started asynchronously 
3) if one fails to publish, it gets put back on the queue for another ~5 minutes or so (the queue class will need new functionality to support this)

Reply:
Ah yeah, you did mention that when we talked. The only issue I see is that I don't think we can correlate back to the version which triggered the action (unless we put more data on the event?) so we can't build the `description` text

Reply:
oh hrmmm, that does complicate it a bit.......  we could add that as a column on the event model, or maybe a serialized column for arbitrary data?

Reply:
Maybe it would be possible to async create the task in ContentView::Publish in a suspended state and the event could try to run the task. Dunno if that's possible out of the box